### ISSUE TITLE
Crash in wireguard-go on rpi

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#117]
Crash wireguard go device stats management. This code uses `atomic.*`, which requires careful struct alignment on 32-bit, per https://golang.org/pkg/sync/atomic/#pkg-note-BUG . Frustratingly, the penalty for getting this wrong is a hard segfault.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11e34]
goroutine 118 [running]:
runtime/internal/atomic.goXadd64(0x256c9ec, 0x94, 0x0, 0x94, 0x5e64a8)
        /usr/lib/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
github.com/tailscale/wireguard-go/device.(*Peer).SendBuffer(0x256c840, 0x2158460, 0x94, 0x94, 0x0, 0x0)
        /home/dave/tail/corp/wireguard-go/device/peer.go:181 +0x158
github.com/tailscale/wireguard-go/device.(*Peer).SendHandshakeInitiation(0x256c840, 0x257c200, 0x0, 0x2)
        /home/dave/tail/corp/wireguard-go/device/send.go:169 +0x284
github.com/tailscale/wireguard-go/device.(*Peer).RoutineNonce(0x256c840)
        /home/dave/tail/corp/wireguard-go/device/send.go:475 +0x198
created by github.com/tailscale/wireguard-go/device.(*Peer).Start
        /home/dave/tail/corp/wireguard-go/device/peer.go:239 +0x270
```

**Comment by danderson** [comment]
Should be fixed, by a combo of tailscale/wireguard-go@52819259e4495fa6f814201df873bdef3578b7ea and tailscale/tailscale@061315acb4c50462ba52393d0f83eeeb9becc6db .

Keeping this bug open though, because my fix is a first-pass hack to get the rpi build working, and adds a bunch of dead weight to `device.Peer`. I'll try something more elegant after some sleep.

**Comment by danderson** [comment]
The change is cleaned up and staged in our "PR to zx2c4" branch, so I'm going to call this closed. It'll land in upstream wireguard-go in our next patchset.
